static/src/javascripts/projects/common/modules/async-call-merger.spec.ts

```diff
@@ -0,0 +1,35 @@
+import Chance from 'chance';
+import { noop } from 'lib/noop';
+import type { TargetCallback } from './async-call-merger';
+import { mergeCalls } from './async-call-merger';
+
+const chance = new Chance();
+const RESULT = chance.string();
+
+describe('AsyncCallMerger', () => {
+	let targetFn: TargetCallback<[string]>;
+	let targetFnWithMerging: TargetCallback<[string]>;
+
+	beforeEach(() => {
+		targetFn = jest.fn((callback) => callback(RESULT));
+		targetFnWithMerging = mergeCalls(targetFn);
+	});
```

Comment:
Totally personal preference (and I know this is keeping the structure we had in the js spec), but I'd prefer to see these inlined into the individual tests. This would result in some duplication but I think it's worth it for the reduced complexity of scoping everything to the test blocks. I think this is especially true because of the use of jest mocks and the possibility of the mock state spilling over into another test if it's not reset. What do you reckon? Totally fine if the answer is "not for this PR" :-)

Reply:
If we’re making it better, we might as well make it more readable!

I’ll declare them as `const` inside each block.